src/Sentry/SentryOptionsExtensions.cs

```diff
@@ -370,6 +370,15 @@ internal static void SetupLogging(this SentryOptions options)
                 options.DiagnosticLogger.LogDebug("Logging enabled with ConsoleDiagnosticLogger and min level: {0}",
                     options.DiagnosticLevel);
             }
+
+            if (options.SettingLocator.GetEnvironment().Equals("production", StringComparison.OrdinalIgnoreCase))
+            {
+                options.DiagnosticLogger.LogWarning("The DiagnosticLogger, specifically on the diagnostic level " +
+                                                    "`debug` (default) can cause performance degradations. " +
+                                                    "Many allocations, and printing to console can slow down your app." +
+                                                    "See https://docs.sentry.io/platforms/dotnet/configuration/diagnostic-logger " +
+                                                    "for more information");
```

Comment:
I think most users don't know what Diagnostic Logger is. All they know is `Debug=true` or not even (because they copied the code we gave them, which includes this).

Worth mentioning that. Possibly simplifying the whole message (details can be in the link).
```suggestion
                options.DiagnosticLogger.LogWarning("Sentry option 'Debug' is set to true while Environment is production. " +
                                                    "Be aware this can cause performance degradation and is not advised. " +
                                                    "See https://docs.sentry.io/platforms/dotnet/configuration/diagnostic-logger " +
                                                    "for more information");
```